eng/pipelines/agent-pool-migration.yml

```diff
@@ -0,0 +1,138 @@
+# The pipeline is mainly used to migrate our agent pool from one version to another version. 
+# Here are how we use the pipeline:
+# 1. What repos are you going to work on for your migration? Put your repos in Parameter 'Repos'. Follow the format in default.
+# 2. The replacement map, e.g We will search all occurances of the strings in Parameter 'MigrateFrom' and replace to the value in 'MigrateTo'. This is regex map, and you can define multiple replacement rules here.
+# 3. The Parameter 'WhatIfPreference' boolean flag allows you to check your changed files before open a PR. Check the box if you don't want to create a PR.
+# 4. The pipeline will generate PRs in specified repos, Parameter 'ShortForMigrateFrom' and 'ShortForMigrateTo' are the short descriptions of the PR purpose.
+#    The PR title will in the format of "Migrate from ${{parameters.ShortForMigrateFrom}} to ${{parameters.ShortForMigrateTo}}"
+#    You can also specify who triggers the changes in Paramter 'GithubAssigneeAlias'. Then your PR will request review from the specified person.
+# 5. Go to the PR link in 'Create pull request' step. Review and merge the PR. Note, the pipeline intented to be aggressive, so take a closer look at the changes.
+#    The PR also has a link pointed back to the pipeline, so people will learn where it comes from and the purpose of the PR.
+trigger: none
+pr: none
+
+parameters:
+- name: Repos
+  type: object
+  default:
+    # The exclude/include paths are valid for ubuntu agents. If use windows, please follow the format like "^.\\sdk\\" 
+    - RepoOwner: Azure
+      RepoName: azure-sdk-tools
+      ExcludePaths: ''
+      IncludePaths: ''
+      Branch: 'main'
+    - RepoOwner: Azure
+      RepoName: azure-sdk
+      ExcludePaths: "^./eng/common/*"
+      IncludePaths: ''
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-android
+      ExcludePaths: "^./sdk/|^./eng/common/|/tests/resources/session-records/"
+      IncludePaths: "^./sdk/.*-matrix.*.json$|^./sdk/.*/.*yml$"
```

Comment:
I think it would be good to document in a comment which rule, `ExcludePaths` or `IncludePaths`, takes precedence.

For example, if I do:

```
ExcludePaths: "^./eng/common/foo.txt"
IncludePaths: "^./eng/common/.*"
```

Will then `/eng/common/foo.txt` be included, or not?

Reply:
This is standard regex. The path itself like the pattern ".\eng\scripts\Repo-File-Content-Replacements.ps1", so if use "^" in your regex, `/eng/common/foo.txt` will not match.

Reply:
If you have 
```
IncludePaths: "/eng/common/.*"
```
Then `/eng/common/foo.txt` would match

Reply:
@sima-zhu I think you have replied here to one of my other comments - note I am asking here about the behavior if given path matches both `ExcludePaths` and `IncludePaths`, not how to interpret the symbols in the regex.